COMETwebapp/ViewModels/Components/ReferenceData/CategoriesTableViewModel.cs

```diff
@@ -111,31 +114,39 @@ public CategoriesTableViewModel(ISessionService sessionService)
         public SourceList<Category> DataSource { get; } = new();
 
         /// <summary>
-        /// Value indicating if the <see cref="ParameterType" /> is deprecated
+        ///    Available <see cref="ReferenceDataLibrary"/>s
         /// </summary>
-        public bool IsAllowedToWrite
-        {
-            get => this.isAllowedToWrite;
-            set => this.RaiseAndSetIfChanged(ref this.isAllowedToWrite, value);
-        }
+        public IEnumerable<ReferenceDataLibrary> ReferenceDataLibraries { get; set; }
 
         /// <summary>
-        /// Method invoked when the component is ready to start, having received its
-        /// initial parameters from its parent in the render tree.
-        /// Override this method if you will perform an asynchronous operation and
-        /// want the component to refresh when that operation is completed.
+        ///    Available <see cref="ClassKind"/>s
         /// </summary>
-        /// <returns>A <see cref="Task" /> representing any asynchronous operation.</returns>
-        public void OnInitializedAsync()
-        {
-            foreach (var siteReferenceDataLibrary in this.sessionService.Session.RetrieveSiteDirectory().SiteReferenceDataLibrary)
-            {
-                this.DataSource.AddRange(siteReferenceDataLibrary.DefinedCategory);
-            }
+        public IEnumerable<ClassKindWrapper> PermissibleClasses { get; set; } = Enum.GetValues<ClassKind>().Select(x => new ClassKindWrapper(x));
 
-            this.UpdateProperties(this.DataSource.Items);
-            this.RefreshAccessRight();
-        }
+        /// <summary>
+        ///    Selected <see cref="ClassKind"/>s
+        /// </summary>
+        public IEnumerable<ClassKindWrapper> SelectedPermissibleClasses { get; set; } = new List<ClassKindWrapper>();
+
+        /// <summary>
+        ///    Selected super <see cref="Category"/>
+        /// </summary>
+        public IEnumerable<Category> SelectedSuperCategories { get; set; } = new List<Category>();
+
+        /// <summary>
+        ///  Indicates if confirmation popup is visible
+        /// </summary>
+        public bool popupVisible { get; set; } = false;
+
+        /// <summary>
+        ///     popum message dialog
+        /// </summary>
+        public string popupDialog { get; set; }
```

Comment:
Rename this property to something coherent with the previous comment. Also, any public property should start with an uppercase